### DISCUSSION TITLE
Welcome to lisp-stat Discussions

### DISCUSSION THREAD

**Discussion by Symbolics** [Lisp-Stat/lisp-stat#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions about Lisp-Stat.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by lukego** [comment]
Glad to find this project! I'm learning more statistics lately and I prefer to program in Lisp so I'm checking this project out with interest.

I'm a bit curious how it relates to the CLML project that also includes a whole lot of statistics code. Just two completely separate projects with overlapping goals? https://github.com/mmaul/clml/tree/master/statistics/src

**Reply by snunez1** [reply]
Separate projects with overlapping goals. CLML was my first choice when looking for projects to base Lisp-Stat on, but one of the requirements was that Lisp-Stat be usable in a commercial setting. CLML's LLGPL (rightly or wrongly) isn't acceptable by most companies, and the author of CLML declined to change it.

Anyway, glad to have you on board. I've got a whole bunch of pending changes I'm getting ready to push, and if there's something you need let me know and I'll see what I can do to get it out there.

**Reply by lukego** [reply]
Thanks for explaining! The immediate hiccup for me is that Lisp-Stat doesn't seem to be installable from Quicklisp at the moment. I'm just grabbing from Github for now. The most immediately interesting package for me is the Vega-Lite integration.

Too bad about the licensing issue with CLML. I haven't read the license and contributor agreement for Lisp-Stat properly yet. I suppose that I have to at some point because I'm not familiar with these specific licenses.

**Reply by snunez1** [reply]
What is the issue in installing from Quicklisp? If you don't mind, will you please open up an issue in the `lisp-stat` repo so I can investigate and track it?

FYI, I've nearly completed a refactor of data frame (i.e. version 2.0) and am doing Vega-Lite next, so if you have an interest in this area your input will be welcome. Specifically I want to move from an a-list definition to a p-list form, somewhat along the lines of how ASDF systems are defined.

**Comment by lukego** [comment]
> What is the issue in installing from Quicklisp?

It actually works! This turned out to be user error and my Quicklisp distribution was not up to date. Thanks for telling me that it's supposed to work :). Great, I will be able to take it for a spin now.

I'm very curious about what you're hacking with data.frame and vega-lite. Is there a good way to follow that work and be part of the conversation?

**Reply by snunez1** [reply]
Data-frame 2.0 is nearly complete; testing/debugging/documenting now, so there's not any discussion points for the current iteration. For new features, I'll put ideas into issues in the data-frame repository for better visibility.

The Vega-Lite work hasn't begun yet, so it's a good time. I'll put the 2-3 main ideas I have for the next version into the repo issues so they can be reviewed. If there's anything you notice in the current version of Vega-Lite, please feel free to open up an issue and we'll see if I can work get the improvements into this iteration.
